synthesizer/src/block/transaction/bytes.rs

```diff
@@ -38,8 +38,16 @@ impl<N: Network> FromBytes for Transaction<N> {
                 let deployment = Deployment::read_le(&mut reader)?;
                 // Read the fee.
                 let fee = Fee::read_le(&mut reader)?;
+                // Read the admin.
+                let admin = Admin::read_le(&mut reader)?;
+
+                // Ensure the admin is valid for the deployment transaction.
+                if !admin.verify(id) {
```

Comment:
I think we can skip this check here in favor of a check in the `VM::verify_deployment` logic. (since the `Transaction::deserialize` doesn't do this check and we might not want the perf hit)